<?xml version="1.0"?>
<launch>
    <!-- MAVROS posix SITL environment launch script -->
    <arg name="headless" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="paused" default="false"/> 
    <arg name="verbose" default="false"/>
    <arg name="screen" default="screen"/>
    <arg name="ns" default="uav_1"/>
    <arg name="log_file" default="$(arg ns)"/>
    <arg name="gazebo_arg" default=""/>
    <arg name="enable_logging" default="false" />
    <arg name="enable_ground_truth" default="true" />

   


    <node pkg="tf" type="static_transform_publisher" name="tf_001" args="0 0 0 0 0 0  map world 10" />


    <arg name="subscript" default="" />
    <arg name="param_file" default="$(find rrt_explorer)/resource/exploration.yaml" />

    <arg name="tf_frame" default="world" />
    <arg name="pcl_topic" default="pointcloud" />
    <arg name="stl_file_path" default="" />
    <arg name="mesh_resolution" default="1.0" />
    <arg name="resolution" default="0.15" />
    <arg name="visualize_max_z" default="2.5" />
    <arg name="sensor_max_range" default="3.0" />


      <node name="rrt_planner_deep" pkg="rrt_explorer" type="rrt_planner_deep" output="screen">
        <param name="tf_frame" type="string" value="$(arg tf_frame)" />
        <param name="resolution" type="double" value="$(arg resolution)" />
        <param name="visualize_max_z" type="double" value="$(arg visualize_max_z)" />
        <param name="sensor_max_range" type="double" value="$(arg sensor_max_range)" />
        <param name="stl_file_path" type="string" value="$(arg stl_file_path)" />
        <param name="mesh_resolution" type="double" value="$(arg mesh_resolution)" />
        <param name="map_publish_frequency" type="double" value="1.0" />
        <remap from="pointcloud_throttled" to="$(arg pcl_topic)"/>
        <remap from="poseStamped" to="/current_pose"/>
        <remap from="rrt_explorer" to="/rrt_explorer"/>
        <rosparam command="load" file="$(arg param_file)" />
      </node>

          <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rrt_explorer)/rviz/rrt_planner.rviz"/>

  
</launch>
